python/ray/dashboard/dashboard.py

```diff
@@ -721,22 +928,27 @@ def run(self):
         while True:
             time.sleep(1.0)
             replies = {}
-            for node in self.nodes:
-                node_id = node["NodeID"]
-                stub = self.stubs[node_id]
-                reply = stub.GetNodeStats(
-                    node_manager_pb2.GetNodeStatsRequest(), timeout=2)
-                reply_dict = MessageToDict(reply)
-                reply_dict["nodeId"] = node_id
-                replies[node["NodeManagerAddress"]] = reply_dict
-            with self._raylet_stats_lock:
-                for address, reply_dict in replies.items():
-                    self._raylet_stats[address] = reply_dict
-            counter += 1
-            # From time to time, check if new nodes have joined the cluster
-            # and update self.nodes
-            if counter % 10:
-                self.update_nodes()
+
+            try:
+                for node in self.nodes:
+                    node_id = node["NodeID"]
+                    stub = self.stubs[node_id]
+                    reply = stub.GetNodeStats(
+                        node_manager_pb2.GetNodeStatsRequest(), timeout=2)
+                    reply_dict = MessageToDict(reply)
+                    reply_dict["nodeId"] = node_id
+                    replies[node["NodeManagerAddress"]] = reply_dict
+                with self._raylet_stats_lock:
+                    for address, reply_dict in replies.items():
+                        self._raylet_stats[address] = reply_dict
+            except Exception:
```

Comment:
Here we should also think about a more specific exception, which errors are you trying to catch here?

Reply:
I forgot which one was it. I guess I added this to handle some errors happening when I started working on this feature (like a month ago). I will just revert this.